sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_client_base.py

```diff
@@ -49,18 +39,39 @@ def __init__(self, vault_url: str, credential: "AsyncTokenCredential", **kwargs:
             self._client = client
             return
 
-        api_version = kwargs.pop("api_version", None)
-        generated = load_generated_api(api_version, aio=True)
+        api_version = kwargs.pop("api_version", DEFAULT_VERSION)
 
         pipeline = kwargs.pop("pipeline", None)
-        if not pipeline:
-            config = generated.config_cls(credential, **kwargs)
-            config.authentication_policy = AsyncChallengeAuthPolicy(credential)
-            pipeline = _build_pipeline(config, **kwargs)
-
-        # generated clients don't use their credentials parameter
-        self._client = generated.client_cls(credentials="", pipeline=pipeline)
-        self._models = generated.models
+        transport = kwargs.pop("transport", None)
+        http_logging_policy = HttpLoggingPolicy(**kwargs)
+        http_logging_policy.allowed_header_names.update(
+            {
+                "x-ms-keyvault-network-info",
+                "x-ms-keyvault-region",
+                "x-ms-keyvault-service-version"
+            }
+        )
+
+        if not transport:
+            from azure.core.pipeline.transport import AioHttpTransport
```

Comment:
I'm a little concerned about the unlikely scenario of someone passing a value for `pipeline`. That person doesn't need a transport and if they don't have `aiohttp`, they get an exception here. Looks like `if not transport and not pipeline` would work because `KeyVaultClient` appears to anticipate `transport=None`. Then again it seems a very obscure and unlikely scenario, so perhaps it's unnecessary to plan for or comment about it 😸

Reply:
that's fair, I can still update that though